### ISSUE TITLE
Unexpected result on Nonlinear regression synthetic dataset.

### ISSUE THREAD

**Issue by WYBupup** [maksym33/FeatureImportanceDL#6]
I am trying to reproduce the experiment on Nonlinear regression data mentioned in the paper. However, the result is unexpected and the MSE of test set is nearly 1.3 while the result in the paper is 0.163. All the hyperparameter is set according to the supplementary material of the paper.

Following is the data generation code of nonlinear regression.
```
def generate_nonlinear_regression_data(n=100, seed=0):
    np.random.seed(seed)
    X = np.random.randn(n, 10)
    Y = -2 * np.sin(2 * X[:, 0]) + np.maximum(0, X[:, 1]) + X[:, 2] + np.exp(-X[:, 3]) + np.random.randn(n)
    perm_inds = np.random.permutation(n)
    X, Y = X[perm_inds], Y[perm_inds]
    return X, Y
```

And followings are the training curve 

selector-loss
![selector_train_loss](https://user-images.githubusercontent.com/61353633/147925916-dc39bd10-aff6-4a78-b806-438a51b91a58.jpg)

operator-loss-train
![operator_train_loss](https://user-images.githubusercontent.com/61353633/147925957-5d366a0f-1075-4c14-8bff-eae4e036f99c.jpg)

operator-losss-train(m_opt)
![operator_train_loss_mopt](https://user-images.githubusercontent.com/61353633/147926000-d74729b5-effd-4323-a339-87d11420d029.jpg)

operator-loss-val
![operator_val_losss](https://user-images.githubusercontent.com/61353633/147926025-7403d418-c9ee-4c72-acc1-646435297a1a.jpg)

operator-loss-val(m_opt)
![operator_val_loss_mopt](https://user-images.githubusercontent.com/61353633/147926063-2a3eb4d0-2fee-4556-ac9b-f5678bded8db.jpg)

Following is the code of non-linear regression example:

```
# dataset parameters
N_TRAIN_SAMPLES = 512
N_VAL_SAMPLES = 256
N_TEST_SAMPLES = 1024
N_FEATURES = 10
FEATURE_SHAPE = (10,)
dataset_label = "_NONLINEAR_REGRESSION"

# Training Parameters
data_batch_size = 32
mask_batch_size = 32
s = 5
s_p = 2
phase_2_start = 6000
max_batches = 30000
early_stopping_patience = 3000

# generate data
X_tr, Y_tr = generate_nonlinear_regression_data(N_TRAIN_SAMPLES)
X_val, Y_val = generate_nonlinear_regression_data(N_VAL_SAMPLES)
X_test, Y_test = generate_nonlinear_regression_data(N_TEST_SAMPLES)

# create feature selector framework
fs = FeatureSelector(data_shape=FEATURE_SHAPE, unmasked_data_size=s, data_batch_size=data_batch_size,
                     mask_batch_size=mask_batch_size, str_id=dataset_label)
# generate operator
fs.create_dense_operator([100, 50, 25, 1], activation="linear", metrics=None)
# generate selector
fs.create_dense_selector([100, 50, 10, 1])
# set early stop parameter
fs.operator.set_early_stopping_params(starting_epoch=phase_2_start, patience_batches=early_stopping_patience,
                                      minimize=True)
# set mask optimizer
fs.create_mask_optimizer(epoch_condition=phase_2_start, perturbation_size=s_p)

# Train
fs.train_networks_on_data(X_tr, Y_tr, max_batches, val_data=(X_val, Y_val), val_freq=16)

# results
importances, optimal_mask = fs.get_importances(return_chosen_features=True)
optimal_subset = np.nonzero(optimal_mask)
test_performance = fs.operator.test_one(X_test, optimal_mask[None, :], Y_test)
print("Importance: ", importances)
print("Optimal subsets: ", optimal_subset)
print("Test performace mse: ", test_performance[0])
```

I have struggled with this issue for a long time, so I wonder if it is the problem of my data generation or the problem of the example code I have written. By the way, I also do the experiment on the MNIST subset, and it seems to work pretty well.

Thanks for your open-source code.

**Comment by maksym33** [comment]
Hi,
thanks for expressing interest in our article.
We applied the standard data normalization procedures to all datasets, which includes normalizing the `y` range for the regression data. Obviously, the normalization process was the same for all methods so that the errors/results are comparable, which is probably why you keep getting ~10 times bigger error.
Please do not hesitate to ask if you have any other questions, I am also happy that the method worked good on the MNIST data :)
Cheers,
Max

**Comment by WYBupup** [comment]
appreciate your fast reply! After normalizing, the result is nearly the same as the paper.
I will close the issue.
Thank you for the share of the code and work on that paper!
